### DISCUSSION TITLE
need help with SessionProviders & UseSession.

### DISCUSSION THREAD

**Discussion by Tizooo** [nextauthjs/next-auth#12000] [Help] [answered]
Hello, im pretty new to coding and i am currently trying to make a website using nextjs and next-auth! now im facing a issue which i dont know how to solve. i checked the internet and argued with chatGPT but found no good solution.

Issue:
i made a Navbar component which should save a login or logout button depending of if the user is authenticated or not. for this i use this simple code:
```js
// components/Navbar.js
"use client"; // Indicate that this component is client-side

import Link from "next/link";
import { signIn, signOut, useSession } from "next-auth/react"; // Import NextAuth functions

const Navbar = () => {
  const { data: session } = useSession(); // Get session data

  return (
    <nav className={styles.navbar}>
      <ul>
        <li>
          <Link href="/">Home</Link>
        </li>
        <li>
          <Link href="/dashboard">Dashboard</Link>
        </li>
        {session ? (
          <li>
            <button onClick={() => signOut()}>Logout</button> {/* Logout button */}
          </li>
        ) : (
          <li>
            <button onClick={() => signIn()}>Login</button> {/* Login button */}
          </li>
        )}
      </ul>
    </nav>
  );
};

export default Navbar;
```
issue i have here is that nextjs wants me to put that in a SessionProvider which i dont know how to use in this case. The only solution i found online was th use the SessionProvider in the layout.js which means i would need to add "use client" to the layout.js which would make everything clientside and therefore i can not do it like this. can anyone help me to use the SessionProvider within the navbar?

**Comment by wilsonadenuga** [comment]
create a provider file, make it a client component, then put the session provider there, import and use the provider in your layout, that should solve the issue you are facing.

**Comment by wilsonadenuga** [comment, marked as answer]
[showcase_2.zip](https://github.com/user-attachments/files/17296734/showcase_2.zip)

You can check out this example

**Reply by Tizooo** [reply]
ty very much again :D
